AmahiAnywhere/AmahiAnywhere/Presentation/Files/FilesViewController.swift

```diff
@@ -571,8 +576,90 @@ extension FilesViewController: UINavigationControllerDelegate, UIImagePickerCont
     func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
         picker.dismiss(animated: true, completion: nil)
         
-        guard let _ = info[.editedImage] as? UIImage else {
-            return
+        let mediaType = info[UIImagePickerController.InfoKey.mediaType] as! CFString
+        var fileName: String = ""
+        var mimeType: String = ""
+        
+        var url: String = ""
+        if directory != nil {
+            url = ServerApi.shared!.getFileUri(directory!)!.absoluteString
+        }
+        else {
+            url = ServerApi.shared!.getShareUri(share)!.absoluteString
+        }
+        
+        let assetPath = info[UIImagePickerController.InfoKey.referenceURL] as? NSURL
+        
+        if assetPath != nil {
+            let ext = assetPath!.absoluteString?.components(separatedBy: "ext=")[1].lowercased()
+            if ext == "png" || ext == "jpg" || ext == "heic" {
+                mimeType = "image/jpeg"
+            }
+            else {
+                if ext == "mp4" {
+                    mimeType = "video/mp4"
+                }
+                else if ext == "mov" {
+                    mimeType = "video/quicktime"
+                }
+                else if ext == "flv" {
+                    mimeType = "video/x-flv"
+                }
+                else if ext == "avi" {
+                    mimeType = "video/x-msvideo"
+                }
+            }
+        }
+        else {
+            if self.fileTypeNumber == 0 {
+                mimeType = "image/jpeg"
+            }
+            else {
+                mimeType = "video/quicktime"
+            }
+        }
+
+        switch mediaType {
+        case kUTTypeImage:
+            guard let image = info[.editedImage] as? UIImage else {
+                return
+            }
+            fileName = "IMG.jpeg"
+            let data: Data = image.jpegData(compressionQuality: 100)!
+            Network.shared.uploadFile(url, data: data, fileName: fileName, mime: mimeType) { success in
+             if success {
+                 self.showStatusAlert(title: "Image was successfully uploaded", true)
+             }
+             else {
+                 self.showStatusAlert(title: "An error occured while uploading the image", true)
+             }
+                 self.presenter.getFiles(self.share, directory: self.directory)
+            }
+
+            break
+        case kUTTypeMovie:
+            guard let videoURL = info[UIImagePickerController.InfoKey.mediaURL] as? NSURL else {
+                return
+            }
+            do {
+                let data = try Data(contentsOf: videoURL as URL, options: .mappedIfSafe)
+                            print(data)
+                fileName = "VIDEO.mp4"
```

Comment:
Not a good way to name a file to be uploaded. The file name should be as unique as possible I think unless the backend handles this for you.

You can use the timestamp or some other dynamic properties to make it unique.